python/desc/twinkles/twinkles_sky.py

```diff
@@ -0,0 +1,160 @@
+        """
+        # Observation MetaData
+        self.obs_metadata = obs_metadata
+
+        # Constraint on the brightest star: 
+        self.brightestStar_gmag_inCat = brightestStar_gmag_inCat
+        self.brightestStarMag = 'gmag > {}'.format(self.brightestStar_gmag_inCat)
+
+        # Constraint on brightest galaxy
+        self.brightestGal_gmag_inCat = brightestGal_gmag_inCat
+        self.brightestGalMag = 'g_ab > {}'.format(self.brightestGal_gmag_inCat)
+
+        self.availableConnections = availableConnections
+
+        # Lists of component phosim Instance Catalogs and CatalogDBObjects
+        # Stars
+        self.compoundStarDBList = [StarObj, CepheidStarObj]
+        self.compoundStarICList = [PhoSimCatalogPoint, PhoSimCatalogPoint]
+
+        # Galaxies
+        self.compoundGalDBList = [GalaxyBulgeObj, GalaxyDiskObj, GalaxyAgnObj]
+        self.compoundGalICList = [PhoSimCatalogSersic2D, PhoSimCatalogSersic2D,
+                                  TwinklesCatalogZPoint]
+
+        # SN 
+        ## SN catalogDBObject
+        if self.availableConnections is None:
+            self.availableConnections = list()
+            self.snObj = SNDBObj(table=sntable, connection=None)
+            self.availableConnections.append(self.snObj.connection)
+        else:
+            self.snObj = SNDBObj(table=sntable, connection=self.availableConnections[0])
+        
+        self.sn_sedfile_prefix = sn_sedfile_prefix
+
+    def writePhoSimCatalog(self, fileName):
+        """
+        write the phosim instance catalogs of stars, galaxies and supernovae in
+        the Twinkles Sky to fileName
+
+        Parameters
+        ----------
+        fileName : string, mandatory
+            Name of the file to which the phoSim Instance Catalog will be
+            written
+
+        """
+        starCat = CompoundInstanceCatalog(self.compoundStarICList,
+                                          self.compoundStarDBList,
+                                          obs_metadata=self.obs_metadata,
+                                          constraint=self.brightestStarMag)
+
+        starCat._active_connections += self.availableConnections # append already open fatboy connections
+        starCat.phoSimHeaderMap = TwinklesPhoSimHeader
+
+        t_before_starCat = time.time()
+        print("writing starCat ")
+        starCat.write_catalog(fileName, chunk_size=10000)
+        t_after_starCat = time.time()
+
+        galCat = CompoundInstanceCatalog(self.compoundGalICList,
+                                         self.compoundGalDBList,
+                                         obs_metadata=self.obs_metadata,
+                                         constraint=self.brightestGalMag,
+                                         compoundDBclass=sprinklerCompound)
+
+        # galCat._active_connections = starCat._active_connections # pass along already open fatboy connections
+        galCat._active_connections = self.availableConnections
```

Comment:
The `CompoundInstanceCatalog` will open new connections if it has to (i.e. if it has to query a database that does not already have an open connection), by passing in `self.availableConnections` rather than `starCat._active_connections` you are not giving `galCat` access to all of the connections that `starCat` opened.  Again, this is not relevant to us, since all of our connections are to fatboy, but it might be relevant to a user who tries to use different databases for different data sets.

Feel free to ignore this comment if that is not a use case we want to support.